src/connectivity/Connectivity.cpp

```diff
@@ -0,0 +1,370 @@
+#include "connectivity/Connectivity.hpp"
+
+
+void Connectivity::SolveElement2node(const std::vector<E3D::Parser::Element> &elemVector) {
+
+    _nElem = elemVector.size();
+    _element2nodeStart.resize(_nElem + 1, 0);
+    int counter = 0;
+    for (size_t ielem = 0; ielem < _nElem; ielem++) {
+        E3D::Parser::Element elem = elemVector[ielem];
+        for (int const &node : elem.getElemNodes()) {
+            counter++;
+        }
+        _element2nodeStart[ielem + 1] = counter;
+    }
+    _element2node.resize(counter);
+    //_nNode = counter;
+
+    //construire element2nodeStart
+    for (size_t ielem = 0; ielem < _nElem; ielem++) {
+        int localindex = 0;
+        for (int i=_element2nodeStart[ielem]; i < _element2nodeStart[ielem+1]; i++) {
+
+            _element2node[i] = elemVector[ielem].getElemNodes()[localindex];
+            localindex++;
+
+        }
+    }
+
+
+}
+
+void Connectivity::SolveNode2element() {
+
+    //node2element
+    _node2elementStart.resize(_nNode + 1);
+    _node2element.resize(_element2nodeStart[_nElem]);
+
+    int startI;
+    int endI;
+    int nodeI;
+    for (size_t ielem = 0; ielem < _nElem; ielem++) {
+
+
+        startI = _element2nodeStart[ielem];
+        endI = _element2nodeStart[ielem + 1];
+
+        for (size_t i = startI; i < endI; i++) {
+            nodeI = _element2node[i];
+            _node2elementStart[nodeI + 1] += 1;
+        }
+
+    }
+    for (size_t i = 1; i < _node2elementStart.size(); i++) {
+        _node2elementStart[i] += _node2elementStart[i - 1];
+        //printf("%2d ", _node2elementStart[i] );
+    }
+
+    int j;
+    int store[_nNode] = {0};
+    for (size_t ielem = 0; ielem < _nElem; ielem++) {
+        startI = _element2nodeStart[ielem];
+        endI = _element2nodeStart[ielem + 1];
+
+        for (size_t i = startI; i < endI; i++) {
+            nodeI = _element2node[i];
+            j = _node2elementStart[nodeI] + store[nodeI];
+            _node2element[j] = ielem;
+            store[nodeI]++;
+
+        }
+
+    }
+
+
+}
+
+void Connectivity::SolveElement2Element(const std::vector<int> &VTK, int BoundaryFaces) {
+
+
+    _element2faceStart.resize(_nElem + 1);
+    _element2elementStart.resize(_nElem + 1);
+
+    for (size_t i = 1; i < _nElem + 1; i++) {
+        _element2elementStart[i] = _nFacefElement[i - 1];
+        _element2elementStart[i] += _element2elementStart[i - 1];
+        _element2faceStart[i] = _element2elementStart[i];
+    }
+
+
+    _nFace = (BoundaryFaces + _element2elementStart[_nElem]) / 2;
+    std::cout << '\n' << "nFace: " << _nFace << '\n';
+
+
+    _face2element.resize(_nFace * 2, -1);
+    _element2face.resize(_element2elementStart[_nElem], -1);
+    _element2element.resize(_element2elementStart[_nElem], -1);
+    _face2nodeStart.resize(_nFace + 1, 0);
+    _face2node.resize(_nFace * 4, -1);
+
+
+    //creating lhelp_indice vector
+    std::vector<std::vector<int>> _lhelp_indice{{0, 2, 1, 0, 1, 3, 1, 2, 3, 2, 0, 3},
```

Comment:
Are these some hardcoded values ?

Reply:
this is a fixed vector depending on the vtk id of the element. its a convention on the order of the nodes and the faces